CMakeLists.txt

```diff
@@ -50,3 +50,15 @@ endif ()
 
 add_subdirectory("src")
 
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMajorVersion
```

Comment:
I chose `SameMajorVersion` here, which is basically semantic versioning. If the user requests version `1.0.0` it will allow `1.4.1` which is the current version. It will not allow `2.0.0` because that implies breaking API changes. If you request `find_package(Seasocks 1.5.0)` while only `1.4.1` is installed it will abort with:

```
  Could not find a configuration file for package "Seasocks" that is
  compatible with requested version "1.5.0".

  The following configuration files were considered but not accepted:

    /tmp/with_change_prefix/lib/cmake/Seasocks/SeasocksConfig.cmake, version: 1.4.1
```